src/PrestaShopBundle/Controller/Admin/ModuleController.php

```diff
@@ -706,7 +764,28 @@ public function getModuleCartAction($moduleId)
             '@PrestaShop/Admin/Module/Includes/modal_read_more_content.html.twig',
             array(
                 'module' => $moduleToPresent,
+                'level' => $this->authorizationLevel(),
             )
         );
     }
+    /**
+     * Return the type of authorization on module page.
+     *
+     * @return int(integer)
+     */
+    public function authorizationLevel()
+    {
+        switch (true) {
+            case ($this->isGranted(PageVoter::DELETE, 'ADMINMODULESSF_')) :
+                return 4;
+            case ($this->isGranted(PageVoter::UPDATE, 'ADMINMODULESSF_')) :
+                return 3;
+            case ($this->isGranted(PageVoter::CREATE, 'ADMINMODULESSF_')) :
+                return 2;
+            case ($this->isGranted(PageVoter::READ, 'ADMINMODULESSF_')) :
+                return 1;
+            default :
+                return 0;
```

Comment:
Let's define some constants like:

```php
const NONE = 0;
const READ = 1;
const READ_CREATE = 2;
```

etc.

But … a person can create but not edit a product?